### PR TITLE
IntroHistoryView Fist PR

### PR DESCRIPTION
**Pull request by YLee2121**
# IntroHistoryView First PR

## Introduction

Building the very first view of the History Module. There are several views, models, and viewmodels existing inside the file `IntroHistoryView` for now just for the convenience of future refactoring.

---

## 📂Files

> 📃IntroHistoryView

* RoutineRowViewModel

    It provides the mocking data and `edit` and `delete` methods for the mocking data.

* RoutineSummaryModel

    It is a mocking model for the summary data of each routing.

* body

    It will loop through the routine to show the summary data in list.

* IntroHistoryView_ListRow
    
    ```swift
    IntroHistoryView_ListRow(element: RoutineSummaryModel)
    ```
    This is a single row of the list, it will show the summary data, provide the swipe action with `edit` and `delete`, and navigate to `RoutineDetailView` by clicking on the row.

> 📃RoutineDetailView

Next view from `IntroHistoryView`

---

## Screenshot

imageimage

image

### CODE REVIEW

**Comment by YLee2121** [comment]
Fix the PR issue, overlapping PR.

**Comment by cheniwei** [comment]
Cool interfaces and the finger tap emoji.

**Comment by YLee2121** [comment]
> Cool interfaces and the finger tap emoji.

Remember to review the change and approve the review

File: GymBro/Views/HistoryViews/IntroHistoryView.swift [hunk 1]
```diff
@@ -3,18 +3,177 @@
...
+                .foregroundColor(.gray)
+            
+            Spacer()
+            Image(systemName: "chevron.left")
+                .foregroundColor(.gray)
+        }
+        .background(
+            NavigationLink(
+                destination: RoutineDetailView(),
+                label: { EmptyView()}
```
Comment (cheniwei) [inline GymBro/Views/HistoryViews/IntroHistoryView.swift]: Should have a space after ().
`label: { EmptyView() }`

File: GymBro/Views/HistoryViews/IntroHistoryView.swift [hunk 2]
```diff
@@ -3,18 +3,177 @@
...
+}
+
+
+// MARK: COMPONENT
+
+/// Mock Model for RoutineRowViewModel, which is mock too
+class RoutineSummaryModel: Identifiable {
+    
+    let id = UUID().uuidString
+    let date: String
```
Comment (alex308248) [inline GymBro/Views/HistoryViews/IntroHistoryView.swift:56]: Should we declare date as a Date variable?

File: GymBro/Views/HistoryViews/IntroHistoryView.swift [hunk 3]
```diff
@@ -3,18 +3,177 @@
...
+/// The list row sub view for IntroHistoryView
+///
+/// swipteAction, delete and edit
+///
+/// navLink, lead to RoutineDetailView for each
+///
+/// ```
+/// IntroHistoryView_ListRow(element: RoutineSummaryModel)
+/// ```
+struct IntroHistoryView_ListRow: View {
```
Comment (alex308248) [inline GymBro/Views/HistoryViews/IntroHistoryView.swift]: I think seperate the IntroHistoryView_ListRow to another file will be easier to understand.

**Comment by alex308248** [comment]
For now, I think it's a good view, but the list row is a little bit hard to understand at first glance. However, we could keep it for further discussion. That's not a problem.

File: GymBro/Views/HistoryViews/IntroHistoryView.swift (hunk 3 above)
Comment (YLee2121) [reply GymBro/Views/HistoryViews/IntroHistoryView.swift]: will do!

File: GymBro/Views/HistoryViews/IntroHistoryView.swift (hunk 2 above)
Comment (YLee2121) [reply GymBro/Views/HistoryViews/IntroHistoryView.swift:56]: Yes, we need to in the future.

I didn't spend lots of time on research how to use SwiftUI Date() function, so I use String for now just for my convenience. 😅😅

File: GymBro/Views/HistoryViews/IntroHistoryView.swift (hunk 3 above)
Comment (YLee2121) [reply GymBro/Views/HistoryViews/IntroHistoryView.swift]: Fixed!

**Review by cheniwei — APPROVED** [review]
Approved!

**Review by alex308248 — APPROVED** [review]
Approved!

File: GymBro/Views/HistoryViews/IntroHistoryView_ListRow.swift [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            
+            Spacer()
+            Text("\(element.totalExercise) E")
+            Text("\(element.totalSet) S")
+            Text(String(format: "%.2f", element.trainingVolume) + " VI")
+            Image(systemName: "hand.tap")
+                .foregroundColor(.gray)
+            
+            Spacer()
+            Image(systemName: "chevron.left")
```
Comment (YLee2121) [inline GymBro/Views/HistoryViews/IntroHistoryView_ListRow.swift:46]: change an icon

File: GymBro/Views/HistoryViews/IntroHistoryView_ListRow.swift [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                label: { EmptyView() }
+            )
+            .opacity(0)
+        )
+        .swipeActions(edge: .trailing, allowsFullSwipe: false) {
+            Button(
+                action: {
+                    vm.deleteItem(element: element)
+                },
+                label: {
```
Comment (YLee2121) [inline GymBro/Views/HistoryViews/IntroHistoryView_ListRow.swift:61]: change an icon

**Pull request merged**